### ISSUE TITLE
Add documentation for CSI RBD PV encryption

### ISSUE THREAD

**Issue by Madhu-1** [rook/rook#7032]
cephcsi supports RBD PV encryption with  KMS (vault) and custom kubernetes secrets. It's good to document the new Encryption Feature. The same Encryption Feature for CephFS will be added in the future.

- [ ] New configmap creation and mapping it to csi pods (template)
- [ ] Documentation of how to add new KMS encryption configurations
etc.

**Comment by Madhu-1** [comment]
/assign @Rakshith-R

Note:- we have added different types of encryption, we need to cover those also

cc @nixpanic

**Comment by Jiawei0227** [comment]
Gentle ping, any update on this?

**Comment by Jiawei0227** [comment]
Checking the ceph-csi rbd template: https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/template/rbd/csi-rbdplugin.yaml

It seems ceph-csi-encryption-kms-config is missing. Wondering if this is intentional or something?

**Comment by Madhu-1** [comment]
We have two options mount it as a file or create a configmap with predefined names. i have updated the issue description to convert the template changes also.

**Comment by humblec** [comment]
> Checking the ceph-csi rbd template: https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/template/rbd/csi-rbdplugin.yaml
> 
> It seems ceph-csi-encryption-kms-config is missing. Wondering if this is intentional or something?

@Jiawei0227 you are correct, as the template changes were/are not in place, the admin has to create a configmap in the namespace of ceph csi pods in that name with the details. We should have it documented here though. By this issue we can get it addressed. thanks.

**Comment by Jiawei0227** [comment]
I am also getting this:

```
MountVolume.MountDevice failed for volume "pvc-a40bf29f-d625-4a96-ad20-19d6041c776b" : rpc error: code = Internal desc = invalid encryption kms configuration: configmaps "csi-kms-connection-details" is forbidden: User "system:serviceaccount:rook-ceph:rook-csi-rbd-plugin-sa" cannot get resource "configmaps" in API group "" in the namespace "rook-ceph"
```

Seems another thing to fix

**Comment by Jiawei0227** [comment]
> I am also getting this:
> 
> ```
> MountVolume.MountDevice failed for volume "pvc-a40bf29f-d625-4a96-ad20-19d6041c776b" : rpc error: code = Internal desc = invalid encryption kms configuration: configmaps "csi-kms-connection-details" is forbidden: User "system:serviceaccount:rook-ceph:rook-csi-rbd-plugin-sa" cannot get resource "configmaps" in API group "" in the namespace "rook-ceph"
> ```
> 
> Seems another thing to fix

Seems it has been fixed in 1.8.0
